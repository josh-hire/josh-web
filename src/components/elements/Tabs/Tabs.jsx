import { clsx } from '@mantine/core';
import PropTypes from 'prop-types';
import { useCallback, useState } from 'react';
import { noop } from 'Utils/common';
import styles from './styles.module.scss';

export default function Tab({
  data,
  defaultValue,
  value: initialValue,
  onChange,
  className,
}) {

  const [currentValue, setCurrentValue] = useState(initialValue ?? defaultValue);

  const _onChange = useCallback(val=>{
    setCurrentValue(val);
    return onChange && onChange(val);
  },[onChange]);

  return (
    <div className={clsx(styles.tabs, className)}>
      {data.map(({ label, value, badge }, key) => (
        <div
          className={clsx([
            styles.item,
            currentValue === value && styles.active,
          ])}
          key={`tab-${value}-${key}`}
          onClick={() => _onChange(value)}
        >
          <div className={styles.label}>
            {label}
            {badge && <span className={styles.badge}>{badge}</span>}
          </div>
          {(currentValue === value) && <div className={styles.divider} />}
        </div>
      ))}
    </div>
  );
}

Tab.defaultProps = {
  className: '',
  data: [],
  defaultValue: '',
  onChange: noop,
  value: undefined,
};

Tab.propTypes = {
  className: PropTypes.string,
  data: PropTypes.array,
  defaultValue: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string,
};
