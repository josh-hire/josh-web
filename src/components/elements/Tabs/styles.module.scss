@import 'Styles/utilities/mixins';

.tabs {
  background: #FFFFFF;
  border: 1px solid #D9DEE4;
  border-radius: 16px;
  display: flex;
  height: 78px;
  justify-content: space-around;
  .item {
    align-items: center;
    border-radius: 16px;
    color: #52575C;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    margin: 0 1rem;
    padding: 28px 20px 0;
    position: relative;
    transition: 0.5s ease-in-out;
    width: 100%;
    &:hover {
      background-color: #EBF8FF;
      color: #25282B;
      transition: 0.5s ease-in-out;
    }
    .label {
      font-size: 14px;
      font-style: normal;
      font-weight: 400;
      .badge {
        margin-left: 12px;
        background: #EA2036;
        border-radius: 20px;
        color: #FFF;
        font-size: 14px;  
        font-style: normal;
        font-weight: 600;
        padding: 2px 8px;
      }
    }
  }
  .active {
    color: #25282B;
    font-size:18px;
    font-weight: bold;
  }
  .divider {
    background-color: #0F6EB5;
    border-radius: 5px 5px 0px 0px;
    height: 5px;
    margin-top: auto;
    width: 100%;
  }
}

@include for-tablet-portrait {
  .tabs {
    height: 100px;
    .item {
      align-items: center;
      border-radius: 16px;
      color: #52575C; 
      transition: 0.5s ease-in-out;
      &:hover {
        background-color: #EBF8FF;
        color: #25282B;
        transition: 0.5s ease-in-out;
      }
      .label {
        font-size: 14px;
        .badge {
          border-radius: 50%;
          font-size: 10px;
        }
      }
    }
  }
}