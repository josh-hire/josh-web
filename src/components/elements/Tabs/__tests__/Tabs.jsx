import '@testing-library/jest-dom';
import Tabs from '../index';
import ShallowRenderer from 'react-test-renderer/shallow';
import '@testing-library/jest-dom';

const data = [
  {
    label:'test',
    value:'test',
    badge:'test'
  },
  {
    label:'test2',
    value:'test2',
    badge:''
  }
];

describe('src/component/elements/Tabs', ()=>{
  test('Tabs render',()=>{
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Tabs data={data} defaultValue="test" value={undefined} />);
    tree.props.children[0].props.onClick();
    expect(tree).toMatchSnapshot();
  });
});
