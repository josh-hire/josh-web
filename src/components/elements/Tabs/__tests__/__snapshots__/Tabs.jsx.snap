// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/component/elements/Tabs Tabs render 1`] = `
<div
  className="tabs"
>
  <div
    className="item active"
    onClick={[Function]}
  >
    <div
      className="label"
    >
      test
      <span
        className="badge"
      >
        test
      </span>
    </div>
    <div
      className="divider"
    />
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="label"
    >
      test2
      
    </div>
  </div>
</div>
`;
